Generals/Code/Libraries/Source/WWVegas/WW3D2/dx8wrapper.cpp

```diff
@@ -2403,10 +2403,11 @@ SurfaceClass * DX8Wrapper::_Get_DX8_Back_Buffer(unsigned int num)
 {
 	DX8_THREAD_ASSERT();
 
-	IDirect3DSurface8 * bb;
+	IDirect3DSurface8 * bb=NULL;
 	SurfaceClass *surf=NULL;
-	DX8CALL(GetBackBuffer(num,D3DBACKBUFFER_TYPE_MONO,&bb));
-	if (bb)
+	HRESULT hres=S_FALSE;
+	DX8CALL_HRES(GetBackBuffer(num,D3DBACKBUFFER_TYPE_MONO,&bb), hres);
+	if (hres==D3D_OK && bb)
```

Comment:
This condition implies that one of these could be true when the other is false. This is unlikely. It probably would be ok to just test `hres==D3D_OK` and put an assert for `bb!=NULL`.

Or the simplest fix here is to just initialize `bb` to NULL.

Reply:
I checked both, because bb is considered invalid, when hres is not OK.
Basically it's an overly cautious approach to ensure it's not even set to a wrong value.

Reply:
It's ok, but no need to be that cautious. The code now does a bit more than necessary.

Reply:
I've had some bad experiences with oleaut32 using the passed pointer as storage for temporary parameters.
That's why I'd rather trust the result, than the pointer.
So if you want, I can remove the pointer initialization and check, that gets rid of two superfluous operations.

Reply:
Up to you. Either way is fine.